src/dmm.cpp

```diff
@@ -267,6 +331,90 @@ bool DMM::isIioManagerStarted() const
 	return manager->started() && ui->run_button->isChecked();
 }
 
+void DMM::checkAndUpdateGainMode(const std::vector<double> &volts)
+{
+	if (volts.size() < 2) {
+		return;
+	}
+
+	std::vector<QLabel *> gainLabels {ui->currentGainCh1Label, ui->currentGainCh2Label};
+	std::vector<QLabel *> errorLabels {ui->ch1ErrorLabel, ui->ch2ErrorLabel};
+
+	double hi = 0.0;
+	double lo = 0.0;
+	std::tie(lo, hi) = m_m2k_analogin->getRangeLimits(libm2k::analog::PLUS_MINUS_2_5V);
+
+	const double hi_hi = hi + 0.1;
+	const double hi_lo = hi - 0.1;
+	const double lo_hi = lo + 0.1;
+	const double lo_lo = lo - 0.1;
+
+	auto insideDeltaZone = [&](double voltage) {
+		return (voltage > hi_lo && voltage < hi_hi)
+				|| (voltage < lo_hi && voltage > lo_lo);
+	};
+
+	// TODO: can we querry analog in channels count?
+	for (int i = 0; i < 2; ++i) {
```

Comment:
Can't we use **m_adc_nb_channels** or **m_m2k_analogin->getNbChannels()** ?

Reply:
Thanks. Forgot about this methods.